ocaml/src/osd_access.ml

```diff
@@ -300,18 +300,19 @@ class osd_access
        f)
     (fun exn ->
      get_osd_info ~osd_id >>= fun (_, state,_) ->
-
+     let open Asd_protocol.Protocol.Error in
      let add_to_errors =
-       let open Asd_protocol.Protocol.Error in
+
        match exn with
        | End_of_file
        | Exn Assert_failed _ -> false
        | _ -> true
      in
      let should_invalidate_pool =
        match exn with
-       (* TODO fill in exceptions for which we shouldn't invalidate the pool
-        *)
+       | Exn Assert_failed _ -> false
+       | Exn Unknown_operation -> false
+       | Exn Full -> false
```

Comment:
While I agree that `Exn Full` shouldn't invalidate the pool, it should still somehow result in the osd not getting chosen for future writes, which as a result of this change is no longer the case.
So imo this line should be removed (or you'll need to change/add some logic elsewhere).